actions/arteria_delivery_service.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+import requests
+import json
+import time
+
+
+# Needs to be run in a Stackstorm virtualenv
+from st2actions.runners.pythonrunner import Action
+
+
+class ArteriaDeliveryService(Action):
+
+    def query_for_status(self, links):
+        STAGING_SUCCESSFUL = 'staging_successful'
+        STAGING_FAILED = 'staging_failed'
+
+        def update_links(links_results):
+            for link, state in links_results.iteritems():
+                if state == STAGING_SUCCESSFUL or state == STAGING_FAILED:
+                    continue
+                else:
+                    response = requests.get(link)
+                    response_as_json = json.loads(response.content)
+                    links_results[link] = response_as_json["status"]
+
+        def is_in_end_state(link_state):
+            if not link_state:
+                return False
+            else:
+                return link_state == STAGING_SUCCESSFUL or link_state == STAGING_FAILED
+
+        links_results = dict(map(lambda x: (x, None), links))
+
+        while not all(map(is_in_end_state, links_results.values())):
```

Comment:
Perhaps you should watch out for unexpected responses (states) and/or errors from the server side around here. If I'm reading this correctly this could end up in an infinite loop otherwise.